src/bignum.rs

```diff
@@ -1,15 +1,17 @@
 //! The bignum system library.
+use crate::types::Uint256;
 
-use super::*;
-
+/// The low-level interface to the system library. Use the wrapper functions unless you know what
+/// you're doing.
 pub mod native {
     extern "C" {
         pub fn bignum_mul256(a: *const u32, b: *const u32, ret: *const u32);
         pub fn bignum_umulmod256(a: *const u32, b: *const u32, modulo: *const u32, ret: *const u32);
     }
 }
 
-pub fn mul256(a: &Uint256, b: &Uint256) -> Uint256 {
+/// Unsigned 256-bit multiplication.
+pub fn umul256(a: &Uint256, b: &Uint256) -> Uint256 {
```

Comment:
Well this isn't just a documentation update :)

Reply:
The problem I have with the naming is that it goes against the "agreed" API. I'm not sure what's the best.

Reply:
ah right i'll fix that